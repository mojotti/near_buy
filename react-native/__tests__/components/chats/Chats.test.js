import React from 'react';
import { shallow } from 'enzyme';
import configureStore from 'redux-mock-store';

import _Chat from '../../../src/components/chats/Chat';

const middlewares = [];
const mockStore = configureStore(middlewares);
const initialState = {
  currentChatsReducer: {},
  chatMessagesReducer: {
    chatMessages: [],
  },
  authorizationReducer: {
    token: 'foo',
  },
};
const store = mockStore(initialState);

const navigation = {
  state: {
    params: {
      item: {
        seller_id: 0,
        buyer_id: 0,
        id: 0,
      },
    },
  },
};

describe('<_Chat />', () => {
  test('renders correctly', () => {
    const wrapper = shallow(
      <_Chat
        navigation={navigation}
        store={store}
      />
    );
    expect(wrapper).toMatchSnapshot();
  });
});
