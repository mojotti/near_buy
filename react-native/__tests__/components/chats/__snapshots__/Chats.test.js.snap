// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Chat /> renders correctly 1`] = `
<mockConstructor
  isAnimated={true}
  messages={
    Array [
      Object {
        "_id": 1,
        "createdAt": 2016-01-01T00:00:00.000Z,
        "text": "Hello developer",
        "user": Object {
          "_id": 2,
          "avatar": "http://localhost:5000/api/v1.0/0/image0.jpg",
          "name": "React Native",
        },
      },
      Object {
        "_id": 2,
        "createdAt": 2016-01-01T00:00:00.000Z,
        "text": "Hello stranger!",
        "user": Object {
          "_id": 1,
          "avatar": "http://localhost:5000/api/v1.0/0/image0.jpg",
          "name": "Raimo",
        },
      },
    ]
  }
  onSend={[Function]}
  user={
    Object {
      "_id": 1,
    }
  }
/>
`;
